packages/events-example/src/commands/reconcile/index.ts

```diff
@@ -1,188 +1,136 @@
   nile!: NileApi;
+  deployment!: PulumiAwsDeployment;
 
-  async waitOnStack(stack: Stack): Promise<void> {
-    let stackInfo;
-    do {
-      stackInfo = await stack.info();
-      this.debug(stackInfo);
-    } while (stackInfo != undefined && stackInfo?.result !== 'succeeded');
-  }
+  async run(): Promise<any> {
+    const { flags } = await this.parse(Reconcile);
 
-  async getStack(stackName: string, program: PulumiFn): Promise<Stack> {
-    const args: InlineProgramArgs = {
-      stackName,
-      projectName: 'tryhard',
-      program,
-    };
-    const stack = await LocalWorkspace.createOrSelectStack(args);
-    await stack.setConfig('aws:region', { value: 'us-west-2' });
-    return stack;
-  }
+    // Nile setup
+    await this.connectNile(flags);
 
-  async createStack(instance: Instance): Promise<UpResult> {
-    const stack = await this.getStack(instance.id, pulumiProgram(instance));
-    await this.waitOnStack(stack);
-    try {
-      CliUx.ux.action.start(`Creating a stack id=${instance.id}`);
-      return await stack.up({ onOutput: console.log });
-    } finally {
-      CliUx.ux.action.stop();
-    }
-  }
+    // Pulumi setup
+    this.deployment = await PulumiAwsDeployment.create(
+      flags.workspace, 
+      { projectSettings: { name: flags.workspace, runtime: 'nodejs' } },
+      pulumiProgramGenerator,
+      { region: flags.region },
+    )
 
-  async destroyStack(id: string): Promise<DestroyResult> {
-    const stack = await this.getStack(id, pulumiProgram({}));
-    await this.waitOnStack(stack);
-    try {
-      CliUx.ux.action.start(`Destroying a stack id=${id}`);
-      return await stack.destroy({ onOutput: console.log });
-    } finally {
-      CliUx.ux.action.stop();
-    }
-  }
+    // Identify current state of data plane and control plane
+    const stacks = await this.deployment.loadStacks();
+    const instances = await this.loadNileInstances(flags.organization, flags.entity);
 
-  async loadPulumiStacks(): Promise<{ [key: string]: StackSummary }> {
-    const stacks = await (
-      await this.localWorkspace.listStacks()
-    ).reduce(async (accP, stack) => {
-      const acc = await accP;
-      const fullStack = await this.getStack(stack.name, pulumiProgram({}));
-      const info = await fullStack.info();
-      if (info?.kind != 'destroy') {
-        acc[stack.name] = stack;
-        this.debug('adding stack', stack);
-      }
-      return acc;
-    }, Promise.resolve({} as { [key: string]: StackSummary }));
-    this.debug(stacks);
-    return stacks;
-  }
-
-  async loadInstances(
-    organization: string,
-    entity: string
-  ): Promise<{ [key: string]: Instance }> {
-    const instances = (
-      await this.nile.entities.listInstances({
-        org: organization,
-        type: entity,
-      })
-    ).reduce((acc, instance: Instance) => {
-      if (instance) {
-        acc[instance.id] = instance;
-      }
+    if (flags.statusCheckOnly) {
+      console.log({ stacks, instances });
```

Comment:
I believe `this.log` is more accurate. Also I think linting doesn't like this? or at least it shouldn't.

Reply:
Yeah, this leaked in from local debugging. I'll remove it.